### PR TITLE
bpo-32469: Improve representation of the coroutines

### PR DESCRIPTION
**Pull request by tarasmatsyk**
Current representation:
<coroutine object f at 0x7f86c9733790>

A goal is to display more information:
<coroutine at 0x7f86c9733790, running, file "/home/antoine/cpython/default/Lib/logging/__init__.py", line 123, code getLogger>

https://bugs.python.org/issue32469

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept your contribution by verifying you have signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

Unfortunately our records indicate you have not signed the CLA. For legal reasons we need you to sign this before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.


Thanks again to your contribution and we look forward to looking at it!

**Comment by tarasmatsyk** [comment]
Note: SLA is in the progress.

**Comment by pitrou** [comment]
Hi @tarasmatsyk,

I think you misunderstood the enhancement request. What I mean is the repr() of actual generator objects, for example:
```
>>> 
>>> def g():
...     yield 1
... 
>>> g()
<generator object g at 0x7f80eb781ad8>
>>> async def f():
...     await None
... 
>>> f()
<coroutine object f at 0x7f80eb781b80>
```

**Comment by pitrou** [comment]
I'm also not sure what `GeneratorWrapper` is for, it's the first time I see it. @1st1

**Comment by tarasmatsyk** [comment]
I see, anyway I would be happy to see the right solution or if no one comes up with it - I might be able to figure it out on my own. I will wait to get a response regarding `GeneratorWrapper` and will close this PR.

**Comment by 1st1** [comment]
GeneratorWrapper is what generator-like objects (like generators compiled with Cython) are wrapped with to become awaitables.

File: Lib/types.py [hunk 1]
```diff
@@ -233,6 +233,14 @@ def __iter__(self):
             return self.__wrapped
         return self
     __await__ = __iter__
+    def __repr__(self):
+        return '<coroutine at %s, %s, file %s, line %s, code %s>' % (
```
Comment (1st1) [inline Lib/types.py]: We can't just use 'coroutine' here, that would make it harder to debug things. I propose to use 'coroutine-like-object', which is a bit verbose but readable.

**Review by 1st1 — CHANGES_REQUESTED** [review]

**Comment by tarasmatsyk** [comment]
@pitrou, thanks for the hint, I have updated coroutine and generator representation. I am happy to make any other changes if I missed something.

@1st1, yep, that makes much more sense, a requested change has been applied.

**Comment by tarasmatsyk** [comment]
An SLA is finally approved and all required changes have been made. If I missed anything - let me know, would be happy to update.

A code phrase:
I have made the requested changes; please review again.

**Comment by ahcub** [comment]
nice work! 
looking forward to seeing those changes merged

**Pull request closed without merging**